deepchem/models/wandblogger.py

```diff
@@ -0,0 +1,144 @@
+            raise Exception(
+                'No dataset provided: `train_dataset` is not set or is empty. '
+                '`train_dataset` is needed for epoch calculations. '
+            )
+
+        if (metrics is None) or (not metrics):
+            raise Exception(
+                'No metrics are provided: `metrics` is not set or is empty'
+            )
+
+        if eval_logging_strategy != "step" and eval_logging_strategy != "epoch":
+            logger.warning(
+                "Warning: `eval_logging_strategy` needs to be either 'step' or 'epoch'. Defaulting to 'step'."
+            )
+            eval_logging_strategy = "step"
+
+
+        self.datasets = {
+            "train": train_dataset,
+            "eval": eval_dataset
+        }
+        self.train_dataset_size = len(self.datasets["train"])
+        self.metrics = metrics
+        self.eval_logging_strategy = eval_logging_strategy
+
+        self.log_model = log_model
+        self.save_dir = save_dir
+
+        # set wandb init arguments
+        self.wandb_init_params = dict(
+            name=name,
+            project=project,
+            entity=entity,
+            mode=mode,
+            id=id,
+            dir=save_dir,
+            resume=resume,
+            anonymous=anonymous
+        )
+        self.wandb_init_params.update(**kwargs)
+        self.initialized = False
+
+    def setup(self):
+        self.wandb.init(**self.wandb_init_params)
+        self.initialized = True
+
+    def check_other_loggers(self, callbacks):
+        for c in callbacks:
+            if isinstance(c, ValidationCallback):
+                logger.warning(
+                    "Note: You are using both WandbLogger and ValidationCallback. "
+                )
+
+    def calculate_epoch_num(self, current_step):
+        self.steps_per_epoch = math.ceil(self.train_dataset_size / self.wandb.config.batch_size)
+        self.epoch_num = current_step // self.steps_per_epoch
+
+    def log(self, model, extra_data, step):
+
+        all_metrics = dict({})
+
+        #Log Training
+        scores = model.evaluate(self.datasets["train"], self.metrics)
+        scores = {'train/' + k: v for k, v in scores.items()}
+        all_metrics.update(scores)
+
+        #Log Eval dataset (interval dependent)
+        if self.datasets["eval"] is not None:
+            if self.eval_logging_strategy == "step" and (step % self.wandb.config.log_frequency == 0):
+                scores = model.evaluate(self.datasets["eval"], self.metrics)
+                scores = {'eval/' + k: v for k, v in scores.items()}
+                all_metrics.update(scores)
+            elif self.eval_logging_strategy == "epoch" and (step % self.steps_per_epoch == 0):
+                scores = model.evaluate(self.datasets["eval"], self.metrics)
+                scores = {'eval/' + k: v for k, v in scores.items()}
+                all_metrics.update(scores)
+
+        self.wandb.log(all_metrics, step=step, commit=False)
```

Comment:
Combine these 3 calls to log into a single call. I think generally `commit=False` is preferred only when your calls to `.log` are scattered in different sections of the code. Since they are all in the same place here lets just do 1 call to `.log`